src/contracts/core/DelegationManager.sol

```diff
@@ -533,20 +521,21 @@ contract DelegationManager is
         uint256 addedShares,
         uint64 maxMagnitude
     ) internal {
-        uint256 slashingFactor = _getSlashingFactor(staker, strategy, maxMagnitude);
-
         // Ensure that the operator has not been fully slashed for a strategy
-        // and that the staker has not been fully slashed if its the beaconChainStrategy
+        // and that the staker has not been fully slashed if it is the beaconChainStrategy
+        uint256 slashingFactor = _getSlashingFactor(staker, strategy, maxMagnitude);
         require(slashingFactor != 0, FullySlashed());
 
-        // Increment operator shares
-        operatorShares[operator][strategy] += addedShares;
-        emit OperatorSharesIncreased(operator, staker, strategy, addedShares);
-
         // Update the staker's depositScalingFactor
         DepositScalingFactor storage dsf = _depositScalingFactor[staker][strategy];
         dsf.update(existingDepositShares, addedShares, slashingFactor);
         emit DepositScalingFactorUpdated(staker, strategy, dsf.scalingFactor());
+
+        // If the staker is delegated to an operator, update the operator's shares
+        if (isDelegated(staker)) {
```

Comment:
Less clear but more efficient, can just check `operator != address(0)`. Fine for either soln here

Reply:
Agreed... though I'm gonna keep this as is, since SLOAD will get cheaper when reading a slot twice in a txn.

It's nice that we use the same helper throughout the contract. I won't die on this hill though, so if we want to change this i'm game.